packages/liveblocks-react-lexical/src/liveblocks-plugin-provider.tsx

```diff
@@ -62,29 +60,24 @@ function getEditorStatus(
  */
 export function useEditorStatus(): EditorStatus {
   const room = useRoom();
-  const provider = providersMap.get(room.id);
-
-  const [status, setStatus] = useState(getEditorStatus(provider));
+  const [status, setStatus] = useState<EditorStatus>("not-loaded");
 
   useEffect(() => {
-    const provider = providersMap.get(room.id);
-
-    setStatus(getEditorStatus(provider));
-
-    if (provider === undefined) {
-      return;
-    }
-
-    const cb = () => setStatus(getEditorStatus(provider));
-
-    provider.on("sync", cb);
-
-    return () => provider.off("sync", cb);
+    return providerEventSource.subscribe((event) => {
+      if (event.id !== room.id) return;
+      setStatus(event.status);
+    });
```

Comment:
I think this is a useful pattern to turn _any_ event source into a stateful hook easily. You could rewrite this into a custom hook, i.e. so you can use `const status = useEventSource(providerEventSource)` here.